### ISSUE TITLE
[Debugger] Paused Overlay

### ISSUE THREAD

**Issue by digitarald** [firefox-devtools/ux#88]
Overlay landed in [bug 1354679](https://bugzilla.mozilla.org/show_bug.cgi?id=1354679) as MVP.

<img width="481" alt="image" src="https://user-images.githubusercontent.com/8599/62389438-ae509200-b514-11e9-80d4-253ba756ef3d.png">

This issue is to discuss basic UX follow up, not much diverging from what we have to get this out to users quickly but in a polished look & feel.

Constrains:

1. Don't stray too much from existing debugger overlays, as users should be able to quickly identify what is going on – an overlay popping in is jarring enough, especially with many websites overdoing it, so staying close helps anchor the experience.
2. Keep it small to not avoid hiding critical website aspects
3. Keep it vibrant enough to make it easy to identify
4. Should probably build on the existing paused reason shows in DevTools, which shows the type of parsing (event, xhr, etc) and additional parameter (click, mousemove, etc).
- …

cc @jasonLaster

**Comment by digitarald** [comment]
In-product mockup of design closer to photon, using the paused-reason colors that are also in the Debugger:

<img width="475" alt="image" src="https://user-images.githubusercontent.com/8599/62389342-6fbad780-b514-11e9-9229-d0b39abfb40a.png">

**Comment by wisniewskit** [comment]
Will this overlay interfere with the "pick an element" tool?

**Comment by jasonLaster** [comment]
@wisniewskit nope - when you open the picker it goes away

**Comment by digitarald** [comment]
> Will this overlay interfere with the "pick an element" tool?

Another clarification, this is enabled on Nightly now for testing.

**Comment by violasong** [comment]
Your mockup looks great! Maybe we should match Chrome for "Paused in Debugger" wording for the overlay, to make it a bit more clear that this is coming from DevTools.

**Comment by digitarald** [comment]
> Maybe we should match Chrome for "Paused in Debugger" wording for the overlay, to make it a bit more clear that this is coming from DevTools.

I like the idea. Thinking about when users encounter this, in all cases DevTools should be open. Maybe it can re-use the iconography that we come up with in #60?

**Comment by violasong** [comment]
As in a decorative pause icon? That could make sense, though if we want to match Chrome, we might want to just add icons for Resume/Step Over buttons.

**Comment by digitarald** [comment]
> As in a decorative pause icon?

Yes

> we might want to just add icons for Resume/Step Over buttons.

@jasonLaster landed those behind `devtools.debugger.features.overlay-step-buttons`.

**Comment by digitarald** [comment]
Related, we got some feedback: https://bugzilla.mozilla.org/show_bug.cgi?id=1579768 cc @jasonLaster 

One thought would have been to have a debugger preference to disable it; but the pref got removed when we enabled it by default in Nightly.

@violasong any idea on how to tweak the overlay to less obtrusive?

**Comment by fvsch** [comment]
Just noticed that the overlay currently plays nice with "Pick an element…" mode, but not with:
- Color Picker (works, but we don't hide the overlay)
- Measure an element (doesn't work)

> how to tweak the overlay to less obtrusive?

Brainstorming:

- Making the overlay toolbar a tad smaller (its around 36-38px, could be around 28-32 maybe)
- Not having the overlay background, just the toolbar (as an option?)
- Allowing users to move the overlay toolbar to the bottom? (useful if you're working on a header component)

**Comment by fvsch** [comment]
Mockup: a more configurable overlay experience.

![Pause overlay toolbar with menu](https://user-images.githubusercontent.com/243601/64966627-6bb5ff80-d89f-11e9-99ad-f57ea7fba2b4.png)

**Comment by digitarald** [comment]
> Making the overlay toolbar a tad smaller (its around 36-38px, could be around 28-32 maybe)

Yeah, takes a bit of space, https://github.com/firefox-devtools/ux/issues/88#issuecomment-517795698 addressed part of this.

> Mockup: a more configurable overlay experience.

Maybe just a single toggle. Adding interactive UI to the overlays has been proven not easy.

Other ideas:

- Only show the model initially when pause is triggered
- Any time the page is hovered over, don't show the modal overlay
- Highlight the modal on click to remind user of the paused state

**Comment by YousufSSyed** [comment]
Has there been any updates on this? I've come across it now and its very annoying.

**Comment by nchevobbe** [comment]
> Has there been any updates on this? I've come across it now and its very annoying.

Hello, several things were discussed here, so I'm not sure what you're running into?

**Comment by YousufSSyed** [comment]
Well its been almost 4 years since there's been an any updates on it.

**Comment by nchevobbe** [comment]
> Well its been almost 4 years since there's been an any updates on it.

on what? Please can you describe the issue you're running into precisely?

**Comment by YousufSSyed** [comment]
Bro chill. I mean there hasn't been a reply on this overlay issue since 2019, and I've come acrossed it myself. I looked up what to do anything and didn't find any answers but found this issue.

**Comment by JeremyLeland** [comment]
Not the person you are replying to, but I found this thread while trying figure out how to hide the debug overlay in Firefox Nightly 118. (The white overlay is kind of blinding with my otherwise-dark setup, and it makes it hard for me to see the canvas drawing operations that I'm trying to step through).

Supposedly devtools.debugger.features.overlay could be used to hide the overlay in the past, but doesn't seem to now. I logged https://bugzilla.mozilla.org/show_bug.cgi?id=1846825 to track this.

**Comment by izogfif** [comment]
@JeremyLeland and others who might find this thread via googling: the preference `devtools.debugger.features.overlay` was removed in Firefox 115. You have to use Firefox version 71, 72, ... till 114.0.1 where this preference is still present.

**Comment by nchevobbe** [comment]
There's a WIP patch to add the option back https://bugzilla.mozilla.org/show_bug.cgi?id=1865439

**Comment by saidaLachgar** [comment]
## Steps to disable the pausing overlay

1. In the address bar, type `about:config` and press Enter.
2. A warning might appear; click "**Accept the Risk and Continue.**"
3. In the search bar, type: `devtools.debugger.pause-overlay`
4. You should see a preference named `devtools.debugger.pause-overlay`. It is set to `true` by default.
5. Double-click on this preference to toggle it to `false`.

**Comment by izogfif** [comment]
>     4. You should see a preference named `devtools.debugger.pause-overlay`. It is set to `true` by default.

@saidaLachgar What version of Firefox are you using? I don't see it in Firefox 131.0 (64-bit) on Windows.

**Comment by nchevobbe** [comment]
> > ```
> > 4. You should see a preference named `devtools.debugger.pause-overlay`. It is set to `true` by default.
> > ```
> 
> @saidaLachgar What version of Firefox are you using? I don't see it in Firefox 131.0 (64-bit) on Windows.

the preference is `devtools.debugger.features.overlay`

**Comment by Saida-Lachgar** [comment]
`devtools.debugger.features.overlay` works for me

**Comment by Andro-Marian** [comment]
Why is not there an options on settings for "devtools.debugger.features.overlay"
I was struggling to disable that white overlay, is annoying.

**Comment by nchevobbe** [comment]
> Why is not there an options on settings for "devtools.debugger.features.overlay" I was struggling to disable that white overlay, is annoying.

We'll add on in https://bugzilla.mozilla.org/show_bug.cgi?id=1846825

**Comment by nchevobbe** [comment]
[Bug 1846825](https://bugzilla.mozilla.org/show_bug.cgi?id=1846825) is now fixed, will be in Firefox 142 (and in tomorrow's Firefox Nightly)
The overlay can be toggled from the Debugger option menu, unchecking the "Show paused overlay":

<img width="754" height="514" alt="Image" src="https://github.com/user-attachments/assets/6f61abbe-2272-47ab-ad8e-371e21a708c6" />
